daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -242,21 +281,25 @@ private[lf] object Anf {
       case source.SEVal(x) => transform(depth, target.SEVal(x))
 
       case source.SEApp(func, args) =>
-        // It's safe to perform ANF if the func-expression has no effects when evaluated.
-        val safeFunc =
-          func match {
-            // we know that trivially in these two cases
-            case source.SEBuiltin(b) =>
-              val overApp = args.lengthCompare(b.arity) > 0
-              !overApp
-            case _ => false
-          }
-        // It's also safe to perform ANF for applications of a single argument.
-        val singleArg = args.lengthCompare(1) == 0
-        if (safeFunc || singleArg) {
-          transformMultiApp(depth, env, func, args)(transform)
+        if (enableFullAnfTransformation) {
+          transformMultiAppRightToLeft(depth, env, func, args)(transform)
```

Comment:
Short note for other reviewers...

Here is the payoff.

When using the right to left evaluation order, we no longer have a concept of an expression being safe-(or-not)-for-ANF, it is always safe! And so we don't need `transformMultiAppLeftToRightSafely` which is the only constructor of (`SExpr`) `SEAppOnlyFunIsAtomic`, which is the only caller of (`Speedy`) `executeApplication`, which is a chunk of additional complexity in speedy runtime evaluation.